src/app_service/service/ens/service.nim

```diff
@@ -140,16 +146,58 @@ QtObject:
         let transactionType = trx["type"].getStr
         if transactionType == $PendingTransactionTypeDto.RegisterENS or
           transactionType == $PendingTransactionTypeDto.SetPubKey:
-          self.pendingEnsUsernames.incl trx["additionalData"].getStr
+          let chainId = trx["network_id"].getInt
+          let ensUsername = trx["additionalData"].getStr
+          let dto = EnsUsernameDto(chainId: chainId, username: ensUsername)
+          self.pendingEnsUsernames.incl(dto)
 
-  proc getMyPendingEnsUsernames*(self: Service): seq[string] =
+  proc getMyPendingEnsUsernames*(self: Service): seq[EnsUsernameDto] =
     for i in self.pendingEnsUsernames.items:
       result.add(i)
 
-  proc getAllMyEnsUsernames*(self: Service, includePending: bool): seq[string] =
-    result = self.settingsService.getEnsUsernames()
-    if(includePending):
-      result.add(self.getMyPendingEnsUsernames())
+  proc getAllMyEnsUsernames*(self: Service, includePending: bool): seq[EnsUsernameDto] =
+
+    var response: JsonNode
+    try:
+      response = status_ens.getEnsUsernames().result
+    except Exception as e:
+      error "error occurred", procName="getAllMyEnsUsernames", msg = e.msg
+      return
+
+    if (response.kind != JArray):
+      warn "expected response is not a json object", procName="getAllMyEnsUsernames"
+      return
+
+    for jsonContact in response:
+      result.add(jsonContact.toEnsUsernameDto())
+
+    if (includePending):
+      for dto in self.getMyPendingEnsUsernames():
+        result.add(dto)
+
+  proc add*(self: Service, chainId: int, username: string): bool =
```

Comment:
why do we need public api for add / remove? 
This should only be handle via register/set public and release?

Reply:
`connectOwnedUsername` is it because u reboot your status folder?
`remove` is it to just not display a ens?